### PR TITLE
refactor: Remove unused code in app layout

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Remove unused code. See comments inline with more details

Related links, issue #, if available: n/a

### How has this been tested?

Locally and my dev-pipeline 

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/drawer.tsx [hunk 1]
```diff
@@ -63,24 +57,15 @@ export function Drawer({
   ariaLabels,
   children,
   isOpen,
-  isHidden,
   isMobile,
-  hasDividerWithSplitPanel,
   onToggle,
   onClick,
   onLoseFocus,
-  extendRight = 0,
```
Comment (just-boris) [inline src/app-layout/drawer.tsx:72]: This value is never set and this makes `styles['opaque-background']` never applied. Removing

File: src/app-layout/drawer.tsx [hunk 2]
```diff
@@ -100,15 +85,10 @@ export function Drawer({
     <div
       className={clsx(styles.drawer, {
         [styles['drawer-closed']]: !isOpen,
-        [styles['drawer-hidden']]: isHidden,
         [styles['drawer-mobile']]: isMobile,
-        [styles['has-divider-with-splitpanel']]: hasDividerWithSplitPanel,
```
Comment (just-boris) [inline src/app-layout/drawer.tsx:105]: `styles['drawer-hidden']` and `styles['has-divider-with-splitpanel']` do not exist. Removing

File: src/app-layout/mobile-toolbar/index.tsx [hunk 3]
```diff
@@ -87,20 +87,20 @@ export function MobileToolbar({
         <MobileToggle
           ref={toggleRefs.navigation}
           type="navigation"
-          className={clsx(sharedStyles['navigation-toggle'], testutilStyles['navigation-toggle'])}
```
Comment (just-boris) [inline src/app-layout/mobile-toolbar/index.tsx:90]: Test utils use the second object. Classes from `sharedStyles` can be removed

File: src/app-layout/styles.scss [hunk 4]
```diff
@@ -132,29 +124,6 @@ $drawer-width: calc(#{awsui.$space-layout-toggle-diameter} + 2 * #{awsui.$space-
   }
 }
 
-.button-toggles-container {
```
Comment (just-boris) [inline src/app-layout/styles.scss:135]: Never applied in Javascript. Removing

File: src/app-layout/index.tsx [hunk 5]
```diff
@@ -405,16 +402,6 @@ const OldAppLayout = React.forwardRef(
...
 
-    const previousContentWidth = usePreviousFrameValue(
-      contentWidthWithSplitPanel - (splitPanelOpenOnTheSide ? splitPanelReportedSize : 0)
-    );
-
-    const contentScaleX = (() => {
-      if (isMobile || !isMotionEnabled || !disableContentPaddings || !previousContentWidth) {
-        return undefined;
-      }
-    })();
```
Comment (just-boris) [inline src/app-layout/index.tsx:416]: This function always returns `undefined`. Removing

File: src/app-layout/notifications/index.tsx [hunk 6]
```diff
@@ -10,25 +9,21 @@ interface NotificationsProps {
   children?: React.ReactNode;
   labels: AppLayoutProps.Labels | undefined;
   topOffset: number | undefined;
-  isMobile: boolean;
 }
 interface NotificationWrapperProps extends NotificationsProps {
   sticky: boolean | undefined;
-  navigationPadding: boolean;
-  toolsPadding: boolean;
-  contentWidthStyles?: React.CSSProperties;
```
Comment (just-boris) [inline src/app-layout/notifications/index.tsx:19]: Never used inside this component. Removing

File: src/app-layout/tools-and-split-panel.tsx [hunk 7]
```diff
@@ -11,33 +11,17 @@ import testutilStyles from './test-classes/styles.css.js';
...
-  contentHeightStyle: ReturnType<typeof useContentHeight>['contentHeightStyle'];
-
   tools: React.ReactNode;
   splitPanel: React.ReactNode;
-
   ariaLabels: AppLayoutProps['ariaLabels'];
-
-  disableContentPaddings: AppLayoutProps['disableContentPaddings'];
   isMobile: boolean;
-  isMotionEnabled: boolean;
```
Comment (just-boris) [inline src/app-layout/tools-and-split-panel.tsx:35]: Never used inside this component. Removing

File: src/app-layout/styles.scss [hunk 8]
```diff
@@ -178,63 +147,10 @@ $drawer-width: calc(#{awsui.$space-layout-toggle-diameter} + 2 * #{awsui.$space-
...
 
-/* stylelint-disable-next-line selector-max-universal */
-.navigation-toggle > * {
-  pointer-events: none;
-}
-
-/* stylelint-disable-next-line selector-max-universal */
-.tools-toggle > * {
-  pointer-events: none;
-}
```
Comment (just-boris) [inline src/app-layout/styles.scss:189]: Obsolete after https://github.com/cloudscape-design/components/pull/648

**Review by michaeldowseza — APPROVED** [review]

**Pull request merged**